### DISCUSSION TITLE
Add an "Integrating Next.js into a Monorepo" section in the Docs.

### DISCUSSION THREAD

**Discussion by floroz** [vercel/next.js#31500] [Ideas]
We have attempted to migrate our web apps from a CRA/Custom Webpack build to Next.js

Unfortunately, we've hit a roadblock trying to understand the restrictions that Next.js puts in place when importing files from external directories outside the root folder.

This was the initial issue I opened: https://github.com/vercel/next.js/issues/31348 and following that I have spent quite a large amount of time navigating the different GH issues and discussion.

The information available is extremely fragmented amongst discussions, issues and PRs. 

This makes it quite challenging for teams to decide that they can safely migrate their applications to Next.js if their project is set up as a monorepo.

I would suggest adding a dedicated section to explain the current challenges, available features, and future adopting plans to all users.

The only mention of "Monorepo" I could find was more targeted at the Vercel setup: https://nextjs.org/docs/migrating/incremental-adoption#micro-frontends-with-monorepos-and-subdomains

Ideally, you'd want to clarify:

- how Next.js integrates into a monorepo
- restrictions/limitations of importing external files
- usage of `experimental.externalDir` and its limitation
- `next-transpile-modules` and when it's recommended to use it (CSS modules support)
- Example of integration with popular tooling (NX, Lerna, Yarn Workspaces)

I am happy to assist/contribute writing some of these paragraphs and demo repository if you think there is any value in this suggestion.

**Comment by KKS1** [comment]
totally agree. I was also looking if we should use Next Js Docs for scaffolding a green field monorepo or use Nx docs for it. And realized there is no mention of Nx/ monorepo arch guidelines on next js docs at all.

**Comment by leerob** [comment]
Thanks for this feedback - agreed this would be helpful.
